src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_params.py

```diff
@@ -79,3 +79,4 @@
 
 register_cli_argument('lock', 'name', options_list=('--name', '-n'))
 register_cli_argument('lock create', 'level', options_list=('--lock-type', '-t'), required=True, **enum_choice_list(LockLevel))
+register_cli_argument('lock update', 'level', options_list=('--lock-type', '-t'), required=True, **enum_choice_list(LockLevel))
```

Comment:
These two lines should be consolidated:
```Python
register_cli_argument('lock', 'level', options_list=('--lock-type', '-t'), **enum_choice_list(LockLevel)) 
```

Essentially, you scope it to "lock" and remove the "requiredness". The CLI will determine this automatically by examining the respective method signatures.